sdks/python/apache_beam/runners/worker/sdk_worker_main.py

```diff
@@ -239,24 +241,24 @@ def _parse_pipeline_options(options_json):
   return PipelineOptions.from_dictionary(_load_pipeline_options(options_json))
 
 
-def _get_state_cache_size(experiments):
-  """Defines the upper number of state items to cache.
-
-  Note: state_cache_size is an experimental flag and might not be available in
-  future releases.
+def _get_state_cache_size_bytes(options):
+  """Return the maximum size of state cache in bytes.
 
   Returns:
-    an int indicating the maximum number of megabytes to cache.
+    an int indicating the maximum number of bytes to cache.
       Default is 100 MB
   """
-
+  max_cache_memory_usage_mb = options.view_as(
+      WorkerOptions).max_cache_memory_usage_mb
+  # to maintain backward compatibility
+  experiments = options.view_as(DebugOptions).experiments or []
   for experiment in experiments:
     # There should only be 1 match so returning from the loop
     if re.match(r'state_cache_size=', experiment):
```

Comment:
Should we warn if users are still using the experiment? There is the possibility of them using both the experiment and the flag leading to undefined behavior.

We will eventually want to rip out the experiment.

Reply:
Added a warning.